### DISCUSSION TITLE
How do I get the training result directly from Python train API

### DISCUSSION THREAD

**Discussion by ithmz** [ultralytics/ultralytics#3996] [Q&A]
I would like to have the training results (mAP, loss, etc.) directly from Python train API
Since I want to draw the chart using matplotlib (not through Tensorboard)
How do I achieve this?

**Comment by glenn-jocher** [comment]
@tsangz189 hello there! 👋

To obtain training results such as mAP, loss, etc., directly from the Python train API, you can access the results stored in the `results` attribute of the returned object after training completes. This attribute is a Pandas DataFrame containing detailed training results for each epoch.

Here's a brief outline of the steps you would follow:

1. Call the `train` function and store the returned object.
2. Access the `results` attribute of this object.
3. Use the data in the DataFrame to create your charts with matplotlib.

Remember, the DataFrame will have columns for each metric, so you can select the specific ones you're interested in for plotting.

Happy charting! 📊
